app/src/main/java/com/example/weather/ui/OverviewFragment.kt

```diff
@@ -20,20 +20,22 @@ import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class OverviewFragment : Fragment() {
-    private lateinit var binding: FragmentOverviewBinding
+    private var fragmentOverviewBinding: FragmentOverviewBinding? = null
```

Comment:
Обычно делают так:

```suggestion
    private var _binding: FragmentOverviewBinding? = null
    private val binding: FragmentOverviewBinding
        get() = requireNotNull(_binding)
```

Тогда можно обращаться к `binding` без проверки на null, когда мы уверены, что он не может быть null

Reply:
А в данном случае мы регистрируем `BroadcastReceiver` в `onResume` и снимаем регистрацию в `onPause`, т.е. данные нам могут прийти только между `onResume` и `onPause`, а в это врмя view != null